packages/arb-token-bridge-ui/src/components/App/App.tsx

```diff
@@ -172,21 +164,70 @@ const Injector = ({ children }: { children: React.ReactNode }): JSX.Element => {
       const l1Address = await l1Signer.getAddress()
       const l2Address = await l2Signer.getAddress()
 
-      try {
-        const l1AddressIsEOA = await addressIsEOA(l1Address, l1Provider)
-        const l2AddressIsEOA = await addressIsEOA(l2Address, l2Provider)
+      const walletType: WalletType = await (async () => {
```

Comment:
I think we can do a couple of things here:
* Instead of doing `addressIsEOA`, let's do `isSmartContract({ address, provider })` (I think this was proposed in the other review) as that's the information we seem to need in most cases.
* I feel like we could move this logic into `useNetworksAndSigners` instead of having it in the global state (we wanna move away from this anyway). So in the case of a successful connection, we could add a field `accountType` which can be eoa, supported sc wallet or unsupported sc wallet.

Reply:
<img width="582" alt="Screenshot 2022-12-07 at 18 27 35" src="https://user-images.githubusercontent.com/47932951/206248861-10868837-7f5f-4afd-96b9-3369defe2b72.png">

In case of sending deposits defaults are already applied if provided `destinationAddress` is undefined